src/model.js

```diff
@@ -424,11 +434,24 @@ const BookshelfModel = ModelBase.extend({
     let columnNames, candidates;
     if (_.isArray(arguments[1])) {
       columnNames = arguments[1];
-      candidates = _.drop(arguments, 2);
     } else {
       columnNames = null;
-      candidates = _.drop(arguments);
     }
```

Comment:
I still think that we cannot change the core Model API.
```js
// This should be possible when "registry" plugin is not used
this.morphTo('imageable', null, Site, Author)
```
It is used by many Bookshelf users. Actually I think it is used also with "registry" plugin. So the best would be that it is not changed for that too. But as there is no way to determine for 2nd argument if the array is the declaration for column names or for Model name and `morphValue`.

So I would suggest to refactor following logic.
```js
if (_.isArray(arguments[1])) {
  columnNames = arguments[1];
} else {
  columnNames = null;
}
if (_.isArray(arguments[1]) || _.isNil(arguments[1])) {
  candidates = _.drop(arguments, 2);
 } else {
  candidates = _.drop(arguments, 1);
}
```
and implement it to look the first array argument to be string as I did indicate [previously](https://github.com/tgriesser/bookshelf/pull/1326/files/c2d916f6c8b84e5436f98b4a354f3157c2fddaa4#r75380359).
```js
if (_.isArray(arguments[1]) && arguments[1].every(_.isString)) {
  columnNames = arguments[1];
  candidates = _.drop(arguments, 2);
} else {
  columnNames = null;
  candidates = _.drop(arguments, 1);
}
```